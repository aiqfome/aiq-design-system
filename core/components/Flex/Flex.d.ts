import React from 'react';
import { DefaultTheme } from 'styled-components';
import { ColorProps, SpaceProps, LayoutProps, FontSizeProps, FontWeightProps, BorderProps, FlexProps, PositionProps, FlexboxProps, TextAlignProps } from 'styled-system';
export interface Props extends SpaceProps, DefaultTheme, LayoutProps, FontSizeProps, FontWeightProps, PositionProps, ColorProps, FlexProps, FlexboxProps, BorderProps, TextAlignProps {
    variant?: 'auto' | 'centralized' | 'fullCentralized';
    isClickable?: boolean;
    color?: string;
    fullHeight?: boolean;
    children?: any;
    className?: string;
    onClick?: (e: any) => void;
    onDragOver?: (e: any) => void;
    onDragEnter?: (e: any) => void;
    onDragLeave?: (e: any) => void;
    onDragEnd?: (e: any) => void;
    onDrop?: (e: any) => void;
    style?: any;
}
export declare const Flex: React.ForwardRefExoticComponent<Props & React.RefAttributes<HTMLDivElement>>;
//# sourceMappingURL=Flex.d.ts.map