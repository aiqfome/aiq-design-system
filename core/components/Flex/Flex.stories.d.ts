import React from 'react';
declare const _default: {
    component: React.ForwardRefExoticComponent<import("./Flex").Props & React.RefAttributes<HTMLDivElement>>;
    title: string;
    decorators: ((...args: any) => any)[];
};
export default _default;
export declare const basic: React.FC;
export declare const centralized: React.FC;
export declare const fullCentralized: React.FC;
export declare const flexFullHeight: React.FC;
export declare const Row: React.FC;
export declare const Column: React.FC;
//# sourceMappingURL=Flex.stories.d.ts.map