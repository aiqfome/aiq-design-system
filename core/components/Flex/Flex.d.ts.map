{"version":3,"file":"Flex.d.ts","sourceRoot":"","sources":["../../../src/components/Flex/Flex.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,MAAM,OAAO,CAAA;AACzB,OAAe,EAAE,YAAY,EAAO,MAAM,mBAAmB,CAAA;AAE7D,OAAO,EAEL,UAAU,EAEV,UAAU,EAEV,WAAW,EAEX,aAAa,EAEb,eAAe,EAEf,WAAW,EAGX,SAAS,EAET,aAAa,EAEb,YAAY,EAEZ,cAAc,EACf,MAAM,eAAe,CAAA;AAEtB,MAAM,WAAW,KACf,SAAQ,UAAU,EAChB,YAAY,EACZ,WAAW,EACX,aAAa,EACb,eAAe,EACf,aAAa,EACb,UAAU,EACV,SAAS,EACT,YAAY,EACZ,WAAW,EACX,cAAc;IAChB,OAAO,CAAC,EAAE,MAAM,GAAG,aAAa,GAAG,iBAAiB,CAAA;IACpD,WAAW,CAAC,EAAE,OAAO,CAAA;IACrB,KAAK,CAAC,EAAE,MAAM,CAAA;IACd,UAAU,CAAC,EAAE,OAAO,CAAA;IACpB,QAAQ,CAAC,EAAE,GAAG,CAAA;IACd,SAAS,CAAC,EAAE,MAAM,CAAA;IAClB,OAAO,CAAC,EAAE,CAAC,CAAC,EAAE,GAAG,KAAK,IAAI,CAAA;IAC1B,UAAU,CAAC,EAAE,CAAC,CAAC,EAAE,GAAG,KAAK,IAAI,CAAA;IAC7B,WAAW,CAAC,EAAE,CAAC,CAAC,EAAE,GAAG,KAAK,IAAI,CAAA;IAC9B,WAAW,CAAC,EAAE,CAAC,CAAC,EAAE,GAAG,KAAK,IAAI,CAAA;IAC9B,SAAS,CAAC,EAAE,CAAC,CAAC,EAAE,GAAG,KAAK,IAAI,CAAA;IAC5B,MAAM,CAAC,EAAE,CAAC,CAAC,EAAE,GAAG,KAAK,IAAI,CAAA;IACzB,KAAK,CAAC,EAAE,GAAG,CAAA;CACZ;AA2CD,eAAO,MAAM,IAAI,8EAmBhB,CAAA"}