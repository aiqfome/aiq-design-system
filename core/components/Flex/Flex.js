"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Flex = void 0;
const react_1 = __importDefault(require("react"));
const styled_components_1 = __importStar(require("styled-components"));
const prop_types_1 = __importDefault(require("prop-types"));
const styled_system_1 = require("styled-system");
const FlexStyled = styled_components_1.default.div `
  ${styled_system_1.space}
  ${styled_system_1.fontSize}
  ${styled_system_1.fontWeight}
  ${styled_system_1.border}
  ${styled_system_1.position}
  ${styled_system_1.color}
  ${styled_system_1.flexbox}
  ${styled_system_1.flex}
  ${styled_system_1.textAlign}
  ${styled_system_1.shadow}

  display: flex;

  &.__flex-variant-centralized {
    justify-content: center;
    align-items: center;
  }

  &.__flex-variant-fullCentralized {
    justify-content: center;
    align-items: center;
    height: 100vh;
  }

  &.__flex-fullHeight {
    height: 100vh;
  }

  ${styled_system_1.layout}

  ${({ isClickable }) => isClickable &&
    styled_components_1.css `
      &:active {
        opacity: 0.8;
        user-select: none;
      }
    `}
`;
exports.Flex = react_1.default.forwardRef((_a, ref) => {
    var { children, color, variant = 'auto', fullHeight, className } = _a, props = __rest(_a, ["children", "color", "variant", "fullHeight", "className"]);
    return (react_1.default.createElement(FlexStyled, Object.assign({ "data-testid": 'flex', className: `${className} __flex-variant-${variant} ${fullHeight && '__flex-fullHeight'}`, ref: ref, color: color }, props), children));
});
exports.Flex.displayName = 'Flex';
exports.Flex.propTypes = {
    color: prop_types_1.default.string,
    variant: prop_types_1.default.oneOf(['auto', 'centralized', 'fullCentralized']),
    fullHeight: prop_types_1.default.bool,
    className: prop_types_1.default.string,
    onClick: prop_types_1.default.func,
    children: prop_types_1.default.node
};
