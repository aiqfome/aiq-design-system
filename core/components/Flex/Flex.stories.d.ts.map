{"version":3,"file":"Flex.stories.d.ts","sourceRoot":"","sources":["../../../src/components/Flex/Flex.stories.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,MAAM,OAAO,CAAA;;;;;;AAMzB,wBAIC;AAED,eAAO,MAAM,KAAK,EAAE,KAAK,CAAC,EAqBzB,CAAA;AAED,eAAO,MAAM,WAAW,EAAE,KAAK,CAAC,EAI/B,CAAA;AAED,eAAO,MAAM,eAAe,EAAE,KAAK,CAAC,EAInC,CAAA;AAED,eAAO,MAAM,cAAc,EAAE,KAAK,CAAC,EAIlC,CAAA;AAED,eAAO,MAAM,GAAG,EAAE,KAAK,CAAC,EAOvB,CAAA;AAED,eAAO,MAAM,MAAM,EAAE,KAAK,CAAC,EAO1B,CAAA"}