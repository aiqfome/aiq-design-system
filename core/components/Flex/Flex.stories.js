"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Column = exports.Row = exports.flexFullHeight = exports.fullCentralized = exports.centralized = exports.basic = void 0;
const react_1 = __importDefault(require("react"));
const addon_knobs_1 = require("@storybook/addon-knobs");
const Flex_1 = require("./Flex");
const Text_1 = require("../Text");
exports.default = {
    component: Flex_1.Flex,
    title: 'Flex',
    decorators: [addon_knobs_1.withKnobs]
};
exports.basic = () => (react_1.default.createElement(Flex_1.Flex, { justifyContent: addon_knobs_1.select('justifyContent', { FlexStart: 'flex-start', FlexEnd: 'flex-end', Center: 'center' }, 'flex-start'), alignItems: addon_knobs_1.select('alignItems', { FlexStart: 'flex-start', FlexEnd: 'flex-end', Center: 'center' }, 'flex-start'), backgroundColor: addon_knobs_1.text('backgroundColor', '#fff'), variant: addon_knobs_1.select('Variant', { Centralized: 'centralized', FullCentralized: 'fullCentralized' }, 'centralized') },
    react_1.default.createElement(Text_1.Text, null, "Flex!")));
exports.centralized = () => (react_1.default.createElement(Flex_1.Flex, { variant: 'centralized' },
    react_1.default.createElement(Text_1.Text, null, "Flex!")));
exports.fullCentralized = () => (react_1.default.createElement(Flex_1.Flex, { variant: 'fullCentralized' },
    react_1.default.createElement(Text_1.Text, null, "Flex!")));
exports.flexFullHeight = () => (react_1.default.createElement(Flex_1.Flex, { fullHeight: true },
    react_1.default.createElement(Text_1.Text, null, "Flex!")));
exports.Row = () => (react_1.default.createElement(Flex_1.Flex, { flexDirection: 'row' },
    react_1.default.createElement(Text_1.Text, null, "Flex!"),
    react_1.default.createElement(Text_1.Text, null, "Flex!"),
    react_1.default.createElement(Text_1.Text, null, "Flex!"),
    react_1.default.createElement(Text_1.Text, null, "Flex!")));
exports.Column = () => (react_1.default.createElement(Flex_1.Flex, { flexDirection: 'column' },
    react_1.default.createElement(Text_1.Text, null, "Flex!"),
    react_1.default.createElement(Text_1.Text, null, "Flex!"),
    react_1.default.createElement(Text_1.Text, null, "Flex!"),
    react_1.default.createElement(Text_1.Text, null, "Flex!")));
