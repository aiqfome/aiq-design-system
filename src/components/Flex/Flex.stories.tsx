import React from 'react'
import { withKnobs, select, text } from '@storybook/addon-knobs'

import { Flex } from './Flex'
import { Text } from '../Text'

export default {
  component: Flex,
  title: 'Flex',
  decorators: [withKnobs]
}

export const basic: React.FC = () => (
  <Flex
    justifyContent={select(
      'justifyContent',
      { FlexStart: 'flex-start', FlexEnd: 'flex-end', Center: 'center' },
      'flex-start'
    )}
    alignItems={select(
      'alignItems',
      { FlexStart: 'flex-start', FlexEnd: 'flex-end', Center: 'center' },
      'flex-start'
    )}
    flexDirection={select(
      'flexDirection',
      { Row: 'row', Column: 'column' },
      'row'
    )}
    backgroundColor={text('backgroundColor', '#fff')}
    variant={select(
      'Variant',
      { Centralized: 'centralized', FullCentralized: 'fullCentralized' },
      'centralized'
    )}
  >
    <Text>Flex!</Text>
  </Flex>
)

export const centralized: React.FC = () => (
  <Flex variant='centralized'>
    <Text>Flex!</Text>
  </Flex>
)

export const fullCentralized: React.FC = () => (
  <Flex variant='fullCentralized'>
    <Text>Flex!</Text>
  </Flex>
)
