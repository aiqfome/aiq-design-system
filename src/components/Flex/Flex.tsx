import React from 'react'
import styled, { css, DefaultTheme } from 'styled-components'
import PropTypes from 'prop-types'
import {
  color,
  ColorProps,
  space,
  SpaceProps,
  layout,
  LayoutProps,
  fontSize,
  FontSizeProps,
  fontWeight,
  FontWeightProps,
  border,
  BorderProps,
  position,
  FlexProps,
  flex,
  PositionProps,
  flexbox,
  FlexboxProps
} from 'styled-system'

export interface Props
  extends SpaceProps,
    DefaultTheme,
    LayoutProps,
    FontSizeProps,
    FontWeightProps,
    PositionProps,
    ColorProps,
    FlexProps,
    FlexboxProps,
    BorderProps {
  variant?: 'auto' | 'centralized' | 'fullCentralized'
  color?: string
  fullHeight?: boolean
  children?: any
  className?: string
  onClick?: (e: any) => void
  onDragOver?: (e: any) => void
  onDragEnter?: (e: any) => void
  onDragLeave?: (e: any) => void
  onDragEnd?: (e: any) => void
  onDrop?: (e: any) => void
  style?: any
}

const flexVariations: { [index: string]: any } = {
  auto: css``,
  centralized: css`
    justify-content: center;
    align-items: center;
  `,
  fullCentralized: css`
    justify-content: center;
    align-items: center;
    height: 100vh;
  `
}

const FlexStyled = styled.div<Props>`
  ${space}
  ${fontSize}
  ${fontWeight}
  ${border}
  ${position}
  ${color}
  ${flexbox}
  ${flex}

  display: flex;
  
  ${({ variant }) => flexVariations[variant || 'auto']}

  ${({ fullHeight }) =>
    !!fullHeight &&
    css`
      height: 100vh;
    `}

  ${layout}
`

export const Flex = React.forwardRef<HTMLInputElement, Props>(
  ({ children, color, ...props }, ref) => {
    return (
      <FlexStyled ref={ref} color={color} {...props}>
        {children}
      </FlexStyled>
    )
  }
)

Flex.displayName = 'Flex'

Flex.propTypes = {
  ref: PropTypes.func,
  color: PropTypes.string,
  onClick: PropTypes.func,
  children: PropTypes.node
}
