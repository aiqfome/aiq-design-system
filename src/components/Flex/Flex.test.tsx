import React from 'react'

import { fireEvent } from '@testing-library/react'

import { render } from '../utils/test/render'

import { Flex } from '../Flex'

describe('Flex', () => {
  it('should have to render without crashing', () => {
    const component = render(<Flex flexDirection='row' />)
    expect(component).toBeTruthy()
  })

  it('should have to render a div', () => {
    const { container } = render(<Flex flexDirection='row' />)
    const div = container.querySelector('div')
    expect(div).toBeInTheDocument()
  })

  it('should have to call onClick when click on div', () => {
    const handleOnClick = jest.fn()
    const { getByTestId } = render(
      <Flex onClick={handleOnClick} flexDirection='row' />
    )

    const div = getByTestId('flex')
    fireEvent.click(div)

    expect(handleOnClick).toHaveBeenCalled()
  })

  it('should have content when has children', () => {
    const { getByText } = render(<Flex flexDirection='row'>Flex!</Flex>)
    const div = getByText('Flex!')
    expect(div).toBeTruthy()
  })

  it('should have class __flex-variant-centralized when variant equal centralized', () => {
    const { getByTestId } = render(<Flex variant='centralized' />)

    const div = getByTestId('flex')

    expect(div).toHaveClass('__flex-variant-centralized')
  })

  it('should have class __flex-variant-fullCentralized when variant equal fullCentralized', () => {
    const { getByTestId } = render(<Flex variant='fullCentralized' />)

    const div = getByTestId('flex')

    expect(div).toHaveClass('__flex-variant-fullCentralized')
  })

  it('should have class __fle-fullHeight when have prop fullHeight', () => {
    const { getByTestId } = render(<Flex fullHeight={true} />)

    const div = getByTestId('flex')

    expect(div).toHaveClass('__flex-fullHeight')
  })

  it('should have the same class that was pass to prop', () => {
    const { getByTestId } = render(<Flex className='test' />)

    const div = getByTestId('flex')

    expect(div).toHaveClass('test')
  })
})
