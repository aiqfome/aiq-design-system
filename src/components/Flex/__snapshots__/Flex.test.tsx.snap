// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must match with the previous snapshot snapshot renders 1`] = `
<div
  className="sc-bdnylx fSUZen"
>
  <span
    className="sc-gtssRu kUMpil"
  >
    Flex component!
  </span>
</div>
`;
